### PR TITLE
AttackBlockCallback: creative fix and javadoc update

### PR DESCRIPTION
**Pull request by Technici4n**
Fix `AttackBlockCallback` not properly sending a packet to the server in creative mode when left click is being held. This expands upon the #1872 patch, which missed this edge case. Thanks to @marchermans who spotted the issue, and insisted enough for me to find it!

I also edited the javadoc to clarify the meaning of the various return values for the event. FYI, here is what the rendered javadoc looks like now:
image

### CODE REVIEW

**Comment by marchermans** [comment]
Oh yes baby. Thanks for finding this. This has been holding up the C&B release for weeks because i thought i had a bug on my end!

**Review by apple502j — COMMENTED** [review]
LGTM, but I think CONSUME_PARTIAL should be changed in another PR.

**Review by apple502j — APPROVED** [review]

File: fabric-events-interaction-v0/src/client/java/net/fabricmc/fabric/mixin/event/interaction/client/ClientPlayerInteractionManagerMixin.java
Comment (modmuss50) [inline fabric-events-interaction-v0/src/client/java/net/fabricmc/fabric/mixin/event/interaction/client/ClientPlayerInteractionManagerMixin.java]: Where is this used?

Comment (Technici4n) [reply fabric-events-interaction-v0/src/client/java/net/fabricmc/fabric/mixin/event/interaction/client/ClientPlayerInteractionManagerMixin.java]: Oh nevermind, it should have been included in my revert commit 😓

**Review by modmuss50 — APPROVED** [review]

**Comment by modmuss50** [comment]
I will merge this when we do the next 1.19.2 release as well. 👍

File: fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/AttackBlockCallback.java
Comment (Juuxel) [inline fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/AttackBlockCallback.java]:
```suggestion
 * <p>On the logical client, the return values have the following meaning:
```
I think we use these terms in other places in FAPI too

File: fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/AttackBlockCallback.java
Comment (Juuxel) [inline fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/AttackBlockCallback.java]:
```suggestion
 * <p>On the logical server, the return values have the following meaning:
```

**Review by Juuxel — APPROVED** [review]

**Pull request merged**
